### DISCUSSION TITLE
Is GetPropertyValue/SetPropertyValue thread-safe?

### DISCUSSION THREAD

**Discussion by zimmy87** [JSBSim-Team/jsbsim#511] [Q&A]
If I call JSBSim::FGFDMExec::Run() on one thread, is it safe to call JSBSim::FGFDMExec::GetPropertyValue() or JSBSim::FGFDMExec::SetPropertyValue() on the same FGFDMExec instance on another thread, or do I need a mutex lock for synchronizing these calls?

**Comment by seanmcleod** [comment]
There is no explicit thread safety code within the JSBSim code base, so you'll need to assume that none of the JSBSim functions are thread-safe.

**Reply by zimmy87** [reply]
Thanks for the quick response @seanmcleod; I'll update my code accordingly
